apps/web/src/components/header/Header.tsx

```diff
@@ -0,0 +1,92 @@
+"use client";
+import React, { useState } from "react";
+import { Box, Typography } from "@mui/material";
+import HeaderLogo from "./HeaderLogo";
+import HeaderSearch from "./HeaderSearch";
+
+export default function Header() {
+  const [isSearchExpanded, setIsSearchExpanded] = useState(false);
+
+  const handleSearchToggle = () => {
+    setIsSearchExpanded(prev => !prev);
+  };
+
+  return (
+    <Box
+      sx={{
+        width: "100%",
+        backgroundColor: isSearchExpanded
+          ? "transparent"
+          : theme => theme.palette.primary.light,
+        padding: isSearchExpanded
+          ? theme => theme.spacing(2)
+          : theme => theme.spacing(2, 4),
+        transition: "background-color 0.3s ease",
+      }}>
+      {isSearchExpanded ? (
+        // Search Bar Expanded View
+        <Box
+          sx={{
+            display: "flex",
+            alignItems: "center",
+            justifyContent: "space-between",
+            width: "100%",
+          }}>
+          <HeaderSearch
+            isExpanded={isSearchExpanded}
+            onCollapse={handleSearchToggle}
+            onExpand={function (): void {
+              throw new Error("Function not implemented.");
+            }}
+          />
+        </Box>
+      ) : (
+        // Default Header View
+        <Box
+          sx={{
+            display: "flex",
+            alignItems: "center",
+            justifyContent: "space-between",
+            width: "100%",
+          }}>
```

Comment:
You could define the shared styles in a BoxStyles object and use that object for each Box component. This will reduce redundancy and make your code cleaner.

Reply:
Fixed it, defined shared styles for components having redundent styles.